### ISSUE TITLE
split node does not split a message (stream) injecting a buffer

### ISSUE THREAD

**Issue by mickym2** [node-red/node-red#3457]
### Current Behavior

If you have a message stream (i.e. a byte stream from a serial node) I would expect that according to the specification the stream will be splitted according a specified buffer pattern. 

![image](https://user-images.githubusercontent.com/58858526/155639067-3f4ca7d3-114e-46bc-bf4a-6a66b74383e2.png)

Edited:
So after testing again -  but the bug is different.

The patter is recognized correctly - but the stream or buffer is not split. It seem that the second part of the split is added to the beginning of the same buffer, instead of creating a new buffer.

### Expected Behavior
Edited:
A new should be created which contains following parts of the buffer

### Steps To Reproduce

Inject a buffer into split node.

### Example flow

```
[
    {
        "id": "6577e958a6bf85ba",
        "type": "split",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "splt": "ll",
        "spltType": "str",
        "arraySplt": 1,
        "arraySpltType": "len",
        "stream": false,
        "addname": "",
        "x": 630,
        "y": 500,
        "wires": [
            [
                "9062aa7ccf2b58e8"
            ]
        ]
    },
    {
        "id": "9062aa7ccf2b58e8",
        "type": "debug",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 790,
        "y": 500,
        "wires": []
    },
    {
        "id": "99129d042106b051",
        "type": "inject",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 340,
        "y": 620,
        "wires": [
            [
                "ffb3a402f539e506"
            ]
        ]
    },
    {
        "id": "ffb3a402f539e506",
        "type": "function",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "func": "msg.payload = Buffer.from(\"Goodbye World and Hello World\");\nreturn msg;",
        "outputs": 1,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 490,
        "y": 620,
        "wires": [
            [
                "d9af744d937b3d2b"
            ]
        ]
    },
    {
        "id": "d9af744d937b3d2b",
        "type": "split",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "splt": "[108,108]",
        "spltType": "bin",
        "arraySplt": 1,
        "arraySpltType": "len",
        "stream": true,
        "addname": "",
        "x": 635,
        "y": 620,
        "wires": [
            [
                "85dad1f51af15d1d"
            ]
        ]
    },
    {
        "id": "85dad1f51af15d1d",
        "type": "debug",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 800,
        "y": 620,
        "wires": []
    },
    {
        "id": "05566641e457ba6b",
        "type": "inject",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 320,
        "y": 680,
        "wires": [
            [
                "cd7ecd87081b8923"
            ]
        ]
    },
    {
        "id": "cd7ecd87081b8923",
        "type": "function",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "func": "var str = \"Goodbye World and Hello World\";\narr = [];\n\nfor (var c of str) {\n    var newMsg = {};\n    newMsg.payload = Buffer.from(c);\n    arr.push(newMsg);\n}\n    \nreturn [arr];\n\n\n",
        "outputs": 1,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 480,
        "y": 680,
        "wires": [
            [
                "7d7edbd19f7c4891"
            ]
        ]
    },
    {
        "id": "7d7edbd19f7c4891",
        "type": "split",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "splt": "[108,108]",
        "spltType": "bin",
        "arraySplt": 1,
        "arraySpltType": "len",
        "stream": true,
        "addname": "",
        "x": 630,
        "y": 680,
        "wires": [
            [
                "f7da43b98f05039b"
            ]
        ]
    },
    {
        "id": "f7da43b98f05039b",
        "type": "debug",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 795,
        "y": 680,
        "wires": []
    },
    {
        "id": "b00f33b47239ae33",
        "type": "inject",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 320,
        "y": 740,
        "wires": [
            [
                "6bb6127205e78dea"
            ]
        ]
    },
    {
        "id": "6bb6127205e78dea",
        "type": "function",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "func": "var str = \"Goodbye World and Hello World\";\narr = [];\n\nfor (var c of str) {\n    var newMsg = {};\n    newMsg.payload = Buffer.from(c);\n    arr.push(newMsg);\n}\n    \nreturn [arr];\n\n\n",
        "outputs": 1,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 480,
        "y": 740,
        "wires": [
            [
                "a9f0c00849c39c03"
            ]
        ]
    },
    {
        "id": "549de3a6fddff335",
        "type": "split",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "splt": "[108,108]",
        "spltType": "bin",
        "arraySplt": 1,
        "arraySpltType": "len",
        "stream": true,
        "addname": "",
        "x": 870,
        "y": 740,
        "wires": [
            [
                "d1e25ab5d322d8b2"
            ]
        ]
    },
    {
        "id": "d1e25ab5d322d8b2",
        "type": "debug",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 1030,
        "y": 740,
        "wires": []
    },
    {
        "id": "a9f0c00849c39c03",
        "type": "delay",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "pauseType": "rate",
        "timeout": "5",
        "timeoutUnits": "seconds",
        "rate": "1",
        "nbRateUnits": "1",
        "rateUnits": "second",
        "randomFirst": "1",
        "randomLast": "5",
        "randomUnits": "seconds",
        "drop": false,
        "allowrate": false,
        "outputs": 1,
        "x": 680,
        "y": 740,
        "wires": [
            [
                "549de3a6fddff335"
            ]
        ]
    },
    {
        "id": "0091968195289ceb",
        "type": "inject",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 340,
        "y": 500,
        "wires": [
            [
                "d6e134ee8ba68625"
            ]
        ]
    },
    {
        "id": "d6e134ee8ba68625",
        "type": "function",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "func": "msg.payload = \"Goodbye World and Hello World\";\nreturn msg;",
        "outputs": 1,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 490,
        "y": 500,
        "wires": [
            [
                "6577e958a6bf85ba"
            ]
        ]
    },
    {
        "id": "e35c1892d46a3e3e",
        "type": "split",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "splt": "[108,108]",
        "spltType": "bin",
        "arraySplt": 1,
        "arraySpltType": "len",
        "stream": false,
        "addname": "",
        "x": 630,
        "y": 560,
        "wires": [
            [
                "edcbf9d008118c30"
            ]
        ]
    },
    {
        "id": "edcbf9d008118c30",
        "type": "debug",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 790,
        "y": 560,
        "wires": []
    },
    {
        "id": "d0ecea01d91b579e",
        "type": "inject",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 340,
        "y": 560,
        "wires": [
            [
                "c5b34a9b08d71634"
            ]
        ]
    },
    {
        "id": "c5b34a9b08d71634",
        "type": "function",
        "z": "7b9f1dda6e01a77d",
        "name": "",
        "func": "msg.payload = \"Goodbye World and Hello World\";\nreturn msg;",
        "outputs": 1,
        "noerr": 0,
        "initialize": "",
        "finalize": "",
        "libs": [],
        "x": 490,
        "y": 560,
        "wires": [
            [
                "e35c1892d46a3e3e"
            ]
        ]
    }
]
```

### Environment

- Node-RED version: 2.1.4
- Node.js version: 16.3.2
- npm version: 8.1.2
- Platform/OS: Win 10
- Browser: Google Chrome

**Comment by dceejay** [comment]
Hi - can you ad an example flow that shows this ? I I use a simple flow splitting a buffer from `Hello World` on `ll` I get `He` and then `o WorldHe` as I would expect...
```
[{"id":"05566641e457ba6b","type":"inject","z":"9b742ff9e822cbf5","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":180,"y":330,"wires":[["cd7ecd87081b8923"]]},{"id":"cd7ecd87081b8923","type":"function","z":"9b742ff9e822cbf5","name":"","func":"msg.payload = Buffer.from(\"Hello World\")\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":330,"y":330,"wires":[["7d7edbd19f7c4891"]]},{"id":"7d7edbd19f7c4891","type":"split","z":"9b742ff9e822cbf5","name":"","splt":"[108,108]","spltType":"bin","arraySplt":1,"arraySpltType":"len","stream":true,"addname":"","x":475,"y":330,"wires":[["f7da43b98f05039b"]]},{"id":"f7da43b98f05039b","type":"debug","z":"9b742ff9e822cbf5","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":640,"y":330,"wires":[]}]
```

**Comment by mickym2** [comment]
> Hi - can you add an example flow that shows this ? If I use a simple flow splitting a buffer from `Goodbye World Hello World` on `ll` I get `Goodbye World He` and then `o WorldGoodbye World He` as I would expect... ie only splits on the ll and not the l.
> 
> ```
> [{"id":"05566641e457ba6b","type":"inject","z":"9b742ff9e822cbf5","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":180,"y":330,"wires":[["cd7ecd87081b8923"]]},{"id":"cd7ecd87081b8923","type":"function","z":"9b742ff9e822cbf5","name":"","func":"msg.payload = Buffer.from(\"Goodbye World and Hello World\")\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":330,"y":330,"wires":[["7d7edbd19f7c4891"]]},{"id":"7d7edbd19f7c4891","type":"split","z":"9b742ff9e822cbf5","name":"","splt":"[108,108]","spltType":"bin","arraySplt":1,"arraySpltType":"len","stream":true,"addname":"","x":475,"y":330,"wires":[["f7da43b98f05039b"]]},{"id":"f7da43b98f05039b","type":"debug","z":"9b742ff9e822cbf5","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":640,"y":330,"wires":[]}]
> ```

So thank you for the reply - so I have to admit that my first description was wrong - but there is still a bug with the split node injecting a buffer as input.

The pattern is correctly detected (it happend why I manually tried to simulate a message stream) but the buffer is not split into separate messages when using a buffer. I added your flow with some additional test in the first posting.

If I use a string as input the split node divides the string into 2 or more messages:

![image](https://user-images.githubusercontent.com/58858526/155801498-0eb9fd1d-b391-4626-989d-0504dc06571a.png)


However if I use a buffer - using your example - only one message is created:

![image](https://user-images.githubusercontent.com/58858526/155801715-5b96e071-25fa-48c8-a356-1579fc099e00.png)

The second part is added at the beginning of the same buffer - instead of creating a new one.

**Comment by mickym2** [comment]
OK - I tested it again with a longer stream and it works well. I think - I misinterpreted something of the buffer output not having in mind that the remaining part of a buffer can only be released when a new stream of buffer arrives. So many thanks for your patience and my misunderstanding. I tried it with a longer buffer and it splits the messages also with a buffer as expected:

![image](https://user-images.githubusercontent.com/58858526/155846311-f3a669ff-d8b5-499e-92a1-8349050f111c.png)
